flask_security/decorators.py

```diff
@@ -53,7 +53,11 @@ def _get_unauthorized_view():
             except BuildError:
                 view = None
         utils.do_flash(*utils.get_message('UNAUTHORIZED'))
-        return redirect(view or request.referrer or '/')
+        redirect_to = '/'
+        if request.referrer and not request.referrer.endswith(request.path):
```

Comment:
> `request.full_path.rstrip('?')`

I'm sorry, but how is `request.full_path` without the query string different from `request.path`?

http://flask-.readthedocs.io/en/latest/api/#flask.Request.path (look at the example provided)

I can see a problem if referrer itself has a query string, but that code doesn't seem to solve it.